### PR TITLE
Updated T5 model card with standardized format

### PR DESCRIPTION
**Pull request by ShararehY**
# What does this PR do?

Updated T5 model card with standardized format

Fixes #36979

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

 @stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/t5.md [hunk 1]
```diff
@@ -14,350 +14,99 @@ rendered properly in your Markdown viewer.
...
+tokenizer = AutoTokenizer.from_pretrained("google-t5/t5-base")
+input_ids = tokenizer("translate English to French: The weather is nice today.", return_tensors="pt").to("cuda")
 
->>> print(tokenizer.batch_decode(output_sequences, skip_special_tokens=True))
-['Das Haus ist wunderbar.', 'Ich arbeite gerne in NYC.']
+output = model.generate(**input_ids)
+print(tokenizer.decode(output[0], skip_special_tokens=True))
 ```
 
-Because T5 has been trained with the span-mask denoising objective,
```
Comment (stevhliu) [inline docs/source/en/model_doc/t5.md:292]: Add some notes here:

```md
## Notes

- You can pad the encoder inputs on the left or right because T5 uses relative scalar embeddings.
- T5 models need a slightly higher learning rate than the default used in [`Trainer`]. Typically, values of `1e-4` and `3e-4` work well for most tasks.

**Review by stevhliu — COMMENTED** [review]
Great job, very clean! 👏

**Comment by ShararehY** [comment]
Hi @stevhliu , Thanks for the suggestions! I applied the changes, could you please review it again?

File: docs/source/en/model_doc/t5.md [hunk 2]
```diff
@@ -14,350 +14,99 @@ rendered properly in your Markdown viewer.
...
->>> labels = target_encoding.input_ids
-
->>> # replace padding token id's of the labels by -100 so it's ignored by the loss
->>> labels[labels == tokenizer.pad_token_id] = -100
-
->>> # forward pass
->>> loss = model(input_ids=input_ids, attention_mask=attention_mask, labels=labels).loss
->>> loss.item()
-0.188
+output = model.generate(**input_ids)
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/model_doc/t5.md:73]: This is unresolved :)

**Review by stevhliu — APPROVED** [review]
Looks great, just one tiny nit then we can merge :)

File: docs/source/en/model_doc/t5.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShararehY) [reply docs/source/en/model_doc/t5.md]: I applied the change now.
